yclienttools/common_rules.py

```diff
@@ -26,6 +26,11 @@ def __init__(self, session, yoda_version):
             "1.7", "1.8"] else "1.9"
         self.default_rule_engine = 'irods_rule_engine_plugin-irods_rule_language-instance'
 
+        try:
+            self.client_hints_rules = self.session.client_hints.get("rules", {})
+        except Exception as e:
+            print("Error: {}. Hint: python-irodsclient needs to be version 2.1 or later to support client_hints.".format(e))
```

Comment:
Can we exit if here if an error occurs when retrieving client hints? Otherwise the same exception will re-occur in `call_rule_user_exists`. We can then also remove the retry code in `call_rule_user_exists`...

Reply:
Removed the retry code.